definitions/fenix.toml

```diff
@@ -18,6 +18,26 @@ description = "The number of days in an observation window that clients used the
 select_expression = "COUNT(DISTINCT DATE(submission_timestamp))"
 data_source = "baseline"
 
+
+[metrics.daily_active_users]
+data_source = "baseline"
```

Comment:
This datasource seems to point at [moz-fx-data-shared-prod.org_mozilla_firefox.baseline](https://github.com/mozilla/metric-hub/blob/main/definitions/fenix.toml#L331) which IIUC is just the fenix release channel baseline data. In [past queries](https://sql.telemetry.mozilla.org/queries/91619) we have pointed at `fenix.baseline` which [contains data from all channels](https://mozilla.acryl.io/dataset/urn:li:dataset:(urn:li:dataPlatform:bigquery,mozdata.fenix.baseline,PROD)/Lineage?is_lineage_mode=true&). I believe the intention is to count clients from all release channels... the non-release counts are going to be very small but I think if we want things to align with the KPI we would need to count those channels.

We should get a data engineer to confirm my claim about `org_mozilla_firefox.baseline` only reflecting the release channel. If its true, we should ascertain if there is a reason why metric hub is using this as its source for baseline data, rather than a source that contains all channels.  

(Similarly, I think the active users aggregates tables are based on the fenix last_seen_joined tables, which would also include the non-release channels. Again we should confirm with @lucia-vargas-a or someone else since there has been recent work there).

Reply:
That's a great catch; I misunderstood the dataset syntax. If I understand you correctly, it sounds like we might want to change the `[data_sources.baseline]` definition in each `toml` file from:

```yaml
default_dataset = "org_mozilla_firefox"
```

to this:
```yaml
default_dataset = "fenix"  # or whatever the corresponding product is
```

Reply:
> I believe the intention is to count clients from all channels... the non-release counts are going to be very small but I think if we want things to align with the KPI we would need to count those channels.
I'm surprised the KPI includes non-release clients, I would've thought it would be release-only. 

Leif is right, `moz-fx-data-shared-prod.org_mozilla_firefox` is Fenix release only. `mozdata.fenix` unions data from all channels.

Reply:
yea that's the idea. but I wonder if there is some reason that I don't know about, e.g. for jetstream, that we use  datasets from `org_mozilla_firefox` instead of `fenix`. @danielkberry or @scholtzan any idea? would it break things to make this switch? if not, should we open a separate PR to do that bit first before adding these?

Reply:
So jetstream/opmon fill in the dataset depending on the app the experiment is running on. So a fenix release experiment gets `org_mozilla_firefox` while a beta experiment gets `org_mozilla_firefox_beta`. The _default dataset_ would be release, but if analysis was needed for another channel, it would switch to that channel

Reply:
The dataset isn't hardcoded in the from expression, it's a variable that's filled in at runtime

Reply:
> I'm surprised the KPI includes non-release clients, I would've thought it would be release-only.

Our precedent is to count clients regardless of channel - the unified_metrics table for example queries from the views that union all the channels together (for mobile). [And for desktop there's no filter for the release channel specifically](https://github.com/mozilla/metric-hub/blob/main/definitions/firefox_desktop.toml#L259-L278). 

 Maybe its debatable whether we SHOULD count pre-release, but it seems to be the precedent anyway.

Reply:
@bochocki I wonder if its possible to pass a `dataset = fenix` line into the metric definition so we don't have to change the default

Reply:
So do we need to make changes here? Or leave everything as-is and rely on end-users to specify the correct dataset?

Reply:
Maybe the baseline dataset should be modified to pull from `fenix` instead, since that's what the KPI is based on.

Reply:
Following @danielkberry's comment, maybe it makes sense to change the dataset since in most of the cases where different datasets are used (opmon/jetstream) the dataset will be specified explicitly, but in other cases (somebody copy-pasting the metric definition) it might not be obvious that something other than the default dataset should be used.

Reply:
IMO, we should change metrics to align with the KPI, so if the KPI is using the `fenix` dataset, then as much as possible, our metrics should too. If we change the definition of the `baseline` dataset to point to `fenix`, then we won't have a notion of default dataset, we'll just pull directly from that one. Which I think is fine.

Reply:
> We should get a data engineer to confirm my claim about org_mozilla_firefox.baseline only reflecting the release channel

I can confirm that this dataset only contains release. 


The reason why it isn't `fenix` to begin with is that when the `baseline` dataset was defined `fenix` didn't exist. Back in the days we didn't have these unified datasets that combine all channels, instead we had to explicitly specify the channel-dataset. Unfortunately, that notion is still encoded in jetstream and mozanalysis, so simply changing the definition of `baseline` could potentially cause some problems in jetstream (we'd need to make some changes there first). 

Also, to avoid breaking existing things, we probably shouldn't simply change the `baseline` definition. Instead, I'd recommend to add a `baseline_v2` data source with the new definition.

Reply:
Thanks @scholtzan, that makes a lot of sense. If it sounds good to everyone, I'll add `baseline_v2` datasets in each of the `toml` files that explicitly point to the relevant `<app_name>.baseline` tables.